composeApp/src/androidMain/kotlin/io/github/couchtracker/tmdb/TmdbLanguage.kt

```diff
@@ -0,0 +1,48 @@
+package io.github.couchtracker.tmdb
+
+/**
+ * A language, compatible with TMDB APIs.
+ * Specification is on https://developer.themoviedb.org/docs/languages
+ *
+ * @property language ISO 639-1
+ * @property country ISO 3166-1
+ */
+data class TmdbLanguage(
+    val language: String,
+    val country: String?,
+) {
+    init {
+        check(language.length == 2) { "Invalid language '$language'" }
+        check(language.all { it in 'a'..'z' }) { "Invalid language '$language'" }
+        if (country != null) {
+            check(country.length == 2) { "Invalid country '$country'" }
+            check(country.all { it in 'A'..'Z' }) { "Invalid country '$country'" }
+        }
+    }
+
+    val apiParameter: String
+        get() = if (country != null) {
+            "$language-$country"
+        } else {
+            language
+        }
+
+    fun serialize() = apiParameter
+
+    companion object {
+        val English = TmdbLanguage("en", null)
+
+        /**
+         * Parses a TmdbLanguage. Supported formats are either `language-COUNTRY` or `language`, where
+         * `language` uses ISO 639-1, and `country` uses ISO 3166-1.
+         */
+        @Suppress("MagicNumber")
+        fun parse(serializedValue: String): TmdbLanguage {
+            return when (serializedValue.length) {
+                2 -> TmdbLanguage(serializedValue, null)
+                5 -> TmdbLanguage(serializedValue.substring(0, 2), serializedValue.substring(3, 5))
+                else -> throw IllegalArgumentException("Invalid tmdb language '$serializedValue'")
+            }
```

Comment:
I feel like the slightly better performance doesn't justify the higher complexity of this piece of code (we could just use `split`).

Reply:
Do you have any data that shows otherwise? Nowadays smartphones should be powerful enough that this kind of micro-optimizations shouldn't be needed anymore, at least until a profiler tells us otherwise.

Reply:
This will be called every time we need to read a row from the tmdb db. No need to involve regexp

Reply:
You're thinking of Java. Kotlin has split without regex: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/split.html